request.go

```diff
@@ -658,43 +657,6 @@ func (req *spaceIndexRequest) setIndex(index interface{}) {
 	req.index = index
 }
 
-type authRequest struct {
```

Comment:
It is possible for a user to send multiple authenticate requests on a connection lifetime, so I don't see a reason to make it dial-only, even though we don't have any public API for it yet.

Reply:
> we don't have any public API for it yet

That it is. We can always do it later if someone needs it.

Reply:
I just don't see a reason to move `authRequest` to `dial` since it's just a pure request in nature, like any other request here.

Reply:
This is only used in `dial.go`. It doesn't make much sense to leave this with the rest of the requests.

Reply:
I think it is possible for this authentication request to become public after a some customer demand, so we'll need to move it back. On the other hand, there is no advantages of moving in to dial now.

Reply:
Could we leave it as it is now?

Reply:
No problems, it has been moved back to `request.go`.